server/mdm/apple/apple_mdm.go

```diff
@@ -1465,7 +1465,10 @@ func GenerateOTAEnrollmentProfileMobileconfig(orgName, fleetURL, enrollSecret st
 
 	q := enrollURL.Query()
 	q.Set("enroll_secret", enrollSecret)
-	enrollURL.RawQuery = q.Encode()
+	if idpUUID != "" {
+		q.Set("idp_uuid", idpUUID)
+	}
+	enrollURL.RawQuery = strings.ReplaceAll(q.Encode(), "&", "&amp;") // Only encode & in URL to escape it in XML
```

Comment:
Hmm I think this would be better done in the template XML itself, via the `xml` helper as I did in the POC. Any reason not to? It would take care of properly XML-escaping the full URL, not only the part we know needs escaping.

Reply:
Got it, thanks for the link, did not know it worked like that - will switch it up.